docs/source/developers/docker.rst

```diff
@@ -0,0 +1,197 @@
+
+Running Docker Builds
+=====================
+
+Most of our Linux based continuous integration tasks are decoupled from public
+CI services using docker and docker-compose. Keeping the CI configuration
+minimal makes local reproducibility possible.
+
+Usage
+-----
+
+There are multiple ways to execute the docker based builds. The recommended is
+to use the archery tool:
+
+Installation
+~~~~~~~~~~~~
+
+Requires ``python>=3.5``. It is recommended to install archery in ``editable``
+mode to automatically update the intallation by pulling the arrow repository.
+
+.. code:: bash
+
+    pip install -e dev/archery[docker]
+
+Inspect the available commands and options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+    archery docker --help
+    archery docker run --help
+
+``archery docker run`` tries to provide a similar interface to
+``docker-compose run``, but with builtin support for building hierarchical
+images.
+
+Examples
+~~~~~~~~
+
+**Execute a build:**
+
+.. code:: bash
+
+    archery docker run conda-python
+
+Archery calls the following docker-compose commands:
+
+.. code:: bash
+
+    docker-compose pull --ignore-pull-failures conda-cpp
+    docker-compose build conda-cpp
+    docker-compose pull --ignore-pull-failures conda-python
+    docker-compose build conda-python
+    docker-compose run --rm conda-python
+
+**To disable the image pulling:**
+
+.. code:: bash
+
+    archery docker run --no-cache conda-python
+
+Which translates to:
+
+.. code:: bash
+
+    docker-compose build --no-cache conda-cpp
+    docker-compose build --no-cache conda-python
+    docker-compose run --rm conda-python
+
+**To disable the cache only for the leaf image:**
+
+Useful to force building the development version of a dependency.
+The leaf image is ``conda-python-pandas`` in the example.
+
+.. code:: bash
+
+    PANDAS=master archery docker run --no-cache-leaf conda-python-pandas
+
+Which translates to:
+
+.. code:: bash
+
+    export PANDAS=master
+    docker-compose pull --ignore-pull-failures conda-cpp
+    docker-compose build conda-cpp
+    docker-compose pull --ignore-pull-failures conda-python
+    docker-compose build conda-python
+    docker-compose build --no-cache conda-python-pandas
+    docker-compose run --rm conda-python-pandas
+
+Note that it doens't pull the conda-python-pandas image and disable the cache
+when building it.
+
+``PANDAS`` is a `build parameter <Docker Build Parameters>`_, see the
+defaults in the .env file.
+
+**To entirely skip building the image:**
+
+.. code:: bash
+
+    archery docker run --no-build conda-python
+
+In order to alter the runtime parameters pass docker environment variables to
+the container during its execution:
+
+.. code:: bash
+
+    archery docker run --env CMAKE_BUILD_TYPE=release ubuntu-cpp
+
+See the available C++ in the ``ci/scripts/cpp_build.sh`` script.
+
+**Run the image with custom command:**
+
+Custom docker commands may be passed as the second argument. The following
```

Comment:
second argument? Looks like the fourth to me.

Reply:
Of the `archery docker run` command.